app/controllers/application_controller.rb

```diff
@@ -3,38 +3,32 @@ class ApplicationController < ActionController::Base
   before_filter :add_extra_sentry_metadata
   protect_from_forgery with: :exception
 
-  def self.permit_only_from_prisons
-    before_filter :reject_untrusted_ips!
-  end
-
   def self.permit_only_from_prisons_or_with_key
-    before_filter :reject_untrusted_ips_and_without_key!
-  end
-
-  def self.permit_only_with_key
-    before_filter :reject_without_key!
+    before_filter :reject_without_key_or_trusted_ip!
   end
 
-  def reject!
-    raise ActionController::RoutingError.new('Go away')
+  def reject_without_key_or_trusted_ip!
+    unless valid_auth_key? || permitted_ip?
+      raise ActionController::RoutingError.new('Go away')
```

Comment:
Yes, on both counts. I couldn't think what to replace this crappy copy with. Probably this should be a bog-standard 404.